### PR TITLE
[master < T916] Fix MATCH + LOAD CSV to load csv only once

### PR DESCRIPTION
**Pull request by Josipmrden**
MATCH + LOAD CSV in this PR will be modified in a way that it allows for MATCH or MERGE to yield one value. If there are more values yielded, it will cause a runtime error.

[master < Task] PR
- [x] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [X] Write a release note here
`MATCH and MERGE clauses now can be put before LOAD CSV if they yield only one row. That way, entities can be matched only once prior to reading rows from a file, which speeds up execution.`
- [X] Tag someone from docs team in the comments

### CODE REVIEW

File: src/query/plan/operator.cpp [hunk 1]
```diff
@@ -4724,12 +4722,14 @@ class LoadCsvCursor : public Cursor {
 
     bool input_pulled = input_cursor_->Pull(frame, context);
 
-    // If the input is Once, we have to keep going until we read all the rows,
-    // regardless of whether the pull on Once returned false.
-    // If we have e.g. MATCH(n) LOAD CSV ... AS x SET n.name = x.name, then we
-    // have to read at most cardinality(n) rows (but we can read less and stop
-    // pulling MATCH).
-    if (!input_is_once_ && !input_pulled) return false;
+    if (input_pulled) {
```
Comment (antoniofilipovic) [inline src/query/plan/operator.cpp:4725]: Slight nitpicking but I think you don't need `input_pulled` variable, you can just place `input_cursor_->Pull(frame, context);` in `if` statement

**Review by antoniofilipovic — APPROVED** [review]
One nitpick, otherwise looks good to go

**Comment by Josipmrden** [comment]
@vpavicic

**Comment by vpavicic** [comment]
@Josipmrden  - ok?

To speed up the execution of the LOAD CSV clause, you can now MATCH and MERGE entities prior to reading the rows from a CSV file. Until now, the LOAD CSV clause had to be the first clause in the query.

**Pull request merged**